OrbitSsh/include/OrbitSsh/LocalSocketManager.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_SSH_FORWARD_SOCKET_MANAGER_H_
+#define ORBIT_SSH_FORWARD_SOCKET_MANAGER_H_
+
+#include <memory>
+#include <string>
+
+#include "OrbitSsh/ResultType.h"
+#include "OrbitSsh/Socket.h"
+
+namespace OrbitSsh {
+
+class LocalSocketManager {
+ public:
+  enum State {
+    kNotInitialized,
+    kInitialized,
+    kBound,
+    kListening,
+    kRunning,
+    kShutdownSent,
+    kRemoteDisconnected
+  };
+
+  explicit LocalSocketManager(int local_port,
```

Comment:
I would remove the default and reorder the arguments to make it more familiar. For convenience you can additionally add a factory function which says in its name that is create a LocalSocketManager which only listens on the internal interface.

Reply:
I changed the order, removed the default value and decided the factory function is not necessary.